language/puppet-functions.md

```diff
@@ -7,16 +7,20 @@ Grammar
 ---
 
     FunctionDefinition
-      : 'function' parameters = ParameterList? '{' statements += statements '}'
+      : 'function' name = NAME parameters = ParameterList? ReturnType? '{' statements += statements '}'
```

Comment:
does not really matter, only entries where the name is needed in the text should have an assignment (or where it acts as a clarification of what the value means). Here it is redundant as the rule is already named "ReturnType" and "return_type = ReturnType" does not give any new information. OTOH, "name = NAME" is equally redundant, I can remove that to make it shorter/more readable.